File: 50-Models/Elysio.Models/DTOs/ConversationDTO.cs
namespace Elysio.Models.DTOs;

public record ConversationDTO
{
    public Guid Id { get; set; }
    public DateTimeOffset CreatedAt { get; set; }
    public DateTimeOffset UpdatedAt { get; set; }

    public string Name { get; set; } = string.Empty;
    public Guid RoomId { get; set; }
    public RoomDTO? Room { get; set; } = null!;
    public List<MessageDTO> Messages { get; set; } = [];
}

File: 10-Client/elysio.crew.front/src/models/interfaces/ConversationDTO.ts
import type { RoomDTO } from "../interfaces/RoomDTO";
import type { MessageDTO } from "../interfaces/MessageDTO";

export type ConversationDTO = {
  id: string;
  createdAt: string;
  updatedAt: string;
  name: string;
  roomId: string;
  room?: RoomDTO;
  messages: MessageDTO[];
};
